### PR TITLE
Fixed Goalie Corner Cases

### LINKED ISSUE #2147: Fix Goalie Corner Cases
### Description of the task
There are some corner cases with the goalie: 
* if the goalie transitions into the dribble state, it will not be restrained to the defense area and will chase the ball anywhere
* If a ball teleports behind the goalie, the goalie will bump the ball into the net

### Acceptance criteria

- [ ] fix bugs listed above

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description

Fixing corner cases outlined in #2147 

### Testing Done
added regression tests

### Resolved Issues
resolves #2147 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by Resocram** [comment]
Just curious what is the goalie corner case?

**Comment by jonathanlew** [comment]
> Just curious what is the goalie corner case?

they're described here: https://github.com/UBC-Thunderbots/Software/issues/2147

**Review by amrm3lm — COMMENTED** [review]
lgtm, just one nit

File: src/software/ai/hl/stp/tactic/goalie/goalie_fsm.h [hunk 1]
```diff
@@ -262,54 +229,34 @@ struct GoalieFSM
...
+                                            .orientation();
+
+                PivotKickFSM::ControlParams control_params{
+                    .kick_origin       = clear_origin,
+                    .kick_direction    = clear_direction,
+                    .auto_chip_or_kick = AutoChipOrKick{AutoChipOrKickMode::AUTOCHIP,
+                                                        YEET_CHIP_DISTANCE_METERS},
+                };
+
+                // update the dribble fsm
```
[the remark this answers is not in the document]
Comment (jonathanlew) [reply src/software/ai/hl/stp/tactic/goalie/goalie_fsm.h]: yup fixed

**Review by akhilveeraghanta — APPROVED** [review]
